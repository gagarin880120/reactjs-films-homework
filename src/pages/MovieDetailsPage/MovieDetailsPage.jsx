import React from 'react';
import PropTypes from 'prop-types';
import Header from '../../components/Header/Header';
import MovieInfoContainer from '../../components/MovieInfo';
import MovieListContainer from '../../components/MovieList';
import FooterContainer from '../../components/Footer';
import Spinner from '../../components/Spinner/Spinner';
import ButtonsWrapperContainer from '../../components/ButtonsWrapper';
import styles from './MovieDetailsPage.module.scss';

export default function MovieDetailsPage({ movie, areMoviesLoaded, results }) {
  const style = {
    backgroundImage: movie.backdrop_path
      ? `url(https://image.tmdb.org/t/p/w1280${movie.backdrop_path})`
      : null,
  };

  return (
    <div className={styles.wrapper} style={style}>
      <Header />
      <main className={styles.main}>
        <MovieInfoContainer />
        <div className={styles.buttons}>
          <ButtonsWrapperContainer />
        </div>
      </main>
      <MovieListContainer />
      <div className={styles.loading}>
        {
          !areMoviesLoaded ? <Spinner /> : null
        }
      </div>
      {
        results.length ? <FooterContainer /> : null
      }
    </div>
  );
}

MovieDetailsPage.propTypes = {
  movie: PropTypes.shape({
    backdrop_path: PropTypes.string,
    id: PropTypes.number,
    overview: PropTypes.string,
  }),
  areMoviesLoaded: PropTypes.bool,
  results: PropTypes.instanceOf(Array),
};

MovieDetailsPage.defaultProps = {
  movie: {
    backdrop_path: '',
    id: 0,
    overview: '',
  },
  areMoviesLoaded: false,
  results: [],
};
