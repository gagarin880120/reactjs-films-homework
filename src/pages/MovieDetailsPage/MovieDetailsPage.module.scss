.wrapper {
  font-family: Arial, Helvetica, sans-serif;
  color: #fff;
}

.infoContainer {
  padding: 0 5% 2%;
  background-size: cover;
}

.main {
  margin-top: 35%;
  display: flex;
  justify-content: space-between;
}

.buttonsContainer {
  margin-top: 100px;
  width: 250px;
  display: flex;
  justify-content: space-between;
}

@media screen and (max-width: 600px) {
  .buttonsContainer {
    width: 150px;
  }
}
