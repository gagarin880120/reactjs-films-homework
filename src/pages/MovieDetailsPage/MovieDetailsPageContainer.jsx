import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MovieDetailsPage from './MovieDetailsPage';
import Spinner from '../../components/Spinner/Spinner';
import {
  areMoviesLoadedSelector, resultsSelector, isMovieLoadedSelector,
} from '../../redux/selectors';
import { getMovies, setCurrentAPIRequest } from '../../redux/actions';
import styles from './MovieDetailsPage.module.scss';

export function MovieDetailsPageContainer({
  areMoviesLoaded, results, isMovieLoaded, onPageLoad,
}) {
  return (
    isMovieLoaded
      ? (
        <MovieDetailsPage
          areMoviesLoaded={areMoviesLoaded}
          results={results}
          onPageLoad={onPageLoad}
        />
      ) : (
        <div className={styles.pageLoading}>
          <Spinner />
        </div>
      )
  );
}

export const mapStateToProps = (state) => ({
  areMoviesLoaded: areMoviesLoadedSelector(state),
  results: resultsSelector(state),
  isMovieLoaded: isMovieLoadedSelector(state),
});

export const mapDispatchToProps = (dispatch) => ({
  onPageLoad() {
    dispatch(setCurrentAPIRequest('popular'));
    dispatch(getMovies('popular'));
  },
});

MovieDetailsPageContainer.propTypes = {
  areMoviesLoaded: PropTypes.bool,
  results: PropTypes.instanceOf(Array),
  isMovieLoaded: PropTypes.bool,
  onPageLoad: PropTypes.func,
};

MovieDetailsPageContainer.defaultProps = {
  areMoviesLoaded: false,
  results: [],
  isMovieLoaded: false,
  onPageLoad: null,
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieDetailsPageContainer);
