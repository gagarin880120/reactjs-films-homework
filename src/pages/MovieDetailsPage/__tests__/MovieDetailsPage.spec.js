import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import MovieDetailsPage from '../MovieDetailsPage';

describe('MovieDetailsPage component', () => {
  test('should render without crashing', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<MovieDetailsPage />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
