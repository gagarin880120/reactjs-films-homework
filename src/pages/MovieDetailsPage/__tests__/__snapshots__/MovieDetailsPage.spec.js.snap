// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsPage component should render without crashing 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "backgroundImage": null,
    }
  }
>
  <Header />
  <div
    className="overview"
  >
    <p>
      
    </p>
  </div>
  <main
    className="main"
  >
    <Memo(Connect(MovieInfoContainer)) />
    <div
      className="buttonsContainer"
    >
      <TrailerButtonRect
        id={0}
        onTrailerButtonClick={null}
      />
      <InfoButton
        onInfoButtonClick={[Function]}
      />
    </div>
  </main>
  <Memo(Connect(MovieListContainer)) />
  <div
    className="loading"
  >
    <Spinner />
  </div>
</div>
`;
